src/kbmod_cmdline/kbmod_result_matcher.py

```diff
@@ -0,0 +1,445 @@
+    max_results : int, optional
+        Maximum number of results to process from the file. If None, all results will be processed
+
+    Returns
+    -------
+    pandas.DataFrame
+        A DataFrame containing the matched results with columns for results file, parameters used, uuid, and matched object name.
+    """
+    try:
+        res = Results.read_table(results_file)
+    except Exception as e:
+        print(f"Error reading in results file {results_file}: {e}")
+    if verbose:
+        print(f"Processing {len(res)} results from file: {results_file}")
+    if max_results is not None:
+        if max_results <= 0:
+            raise ValueError("max_results must be a positive integer.")
+        if max_results < len(res):
+            print(f"Limiting processing to the first {max_results} results.")
+            res.table = res.table[:max_results]
+    known_objs_matcher = KnownObjsMatcher(
+        ephem_table,
+        res.mjd_mid,
+        matcher_name="known_matcher",
+        sep_thresh=sep_thresh,
+        time_thresh_s=time_thresh_s,
+        name_col="Name",
+        ra_col=f"ra_{barycentric_dist}",
+        dec_col=f"dec_{barycentric_dist}",
+        mjd_col="mjd_mid",
+    )
+    if verbose:
+        print(f"Finished matching results in {results_file} against known objects.")
+
+    # Get the global WCS from the results for matching.
+    wcs = res.wcs
+    if wcs is None:
+        # Get the global_wcs string from the first row of the ImageCollection as a WCS object.
+        ic = ImageCollection.read(get_ic_from_results_file(results_file))
+        from astropy.wcs import WCS
+
+        wcs = WCS(ic[0]["global_wcs"])
+
+    # Carry out initial matching to known objects and populate the matches column.
+    known_objs_matcher.match(res, wcs)
+
+    # Filter the matches down to results with at least min_obs observations.
+    known_objs_matcher.match_on_min_obs(res, min_obs)
+
+    uuids = []
+    matched_names = []
+    # The number of matched observations for each result and object.
+    matched_obs = []
+    # The propotion of observations for a result that were matched to a known object.
+    obs_ratios = []
+    for row in res:
+        if row["known_matcher"] is not None:
+            for match, obs_mask in row["known_matcher"].items():
+                uuids.append(row["uuid"])
+                matched_names.append(match)
+                # Sum the boolean mask to get the number of matched observations for this result and object.
+                num_matched_obs = sum(obs_mask)
+                matched_obs.append(num_matched_obs)
+                obs_ratios.append(num_matched_obs / len(obs_mask))
+    matching_df = pd.DataFrame(
+        {
+            "results_file": [results_file] * len(uuids),
+            "barycentric_dist": [barycentric_dist] * len(uuids),
+            "sep_thresh": [sep_thresh] * len(uuids),
+            "time_thresh_s": [time_thresh_s] * len(uuids),
+            "min_obs": [min_obs] * len(uuids),
```

Comment:
Since a bunch of this is the same for every line, do we need it in table format? Pandas allows you to set table-wise meta data that we could use instead. Of course it really depends on how you are going to use it.

Actually, I see below you are outputting it as a CSV. So this makes sense.